src/new-client/src/App/Trades/TradesState/filterState.ts

```diff
@@ -5,6 +5,48 @@ import { Trade, trades$ } from "services/trades"
 import { ColField, colFields } from "./colConfig"
 import { createListener, mergeWithKey } from "@react-rxjs/utils"
 
+export type ComparatorType =
+  | "Equals"
+  | "NotEqual"
+  | "Less"
+  | "LessOrEqual"
+  | "Greater"
+  | "GreaterOrEqual"
+  | "InRange"
+
+export const comparatorConfigs: Record<ComparatorType, string> = {
+  Equals: "Equals",
+  NotEqual: "Not equal",
+  Less: "Less than",
+  LessOrEqual: "Less than or equals",
+  Greater: "Greater than",
+  GreaterOrEqual: "Greater than or equals",
+  InRange: "In range",
+}
+
+export interface NumFilterContent {
+  comparator: ComparatorType
+  value1: number | null
+  value2?: number | null
+}
+
+export type DistinctNums = {
```

Comment:
As we move to having different filter types, I think we need to make these types conditional.   Forex, on my branch:

```
export const colFilterMap = mapObject(colConfigs, (val) => val.filterType)

export type DistinctValues = {
  [K in ColField]: typeof colFilterMap[K] extends "set"
  ? Set<Trade[K]> : null
}```

Reply:
Do you want me to combine different filters into one observable? Or you just want to combine different types together?

Reply:
I don't have it completely worked out.  But something like this:
```
export enum DateFilterType {
  equals = "Equals",
  greaterThan = "Greater Than",
  lessThan = "Less Than",
  notEqual = "Not Equal",
  inRange = "In Range",
}

export enum NumberFilterType {
  equals = "Equals",
  greaterThan = "Greater Than",
  greaterThanOrEqual = "Greater Than or Equal",
  lessThan = "Less Than",
  lessThanOrEqual = "Less Than or Equal",
  notEqual = "Not Equal",
  inRange = "In Range",
}

export type DistinctValues = {
  [K in ColField]: typeof colFilterMap[K] extends "set"
  ? Set<Trade[K]> : null
}

export type AppliedFilters = {
  [K in ColField]: typeof colFilterMap[K] extends "date"
    ? DateFilterType
    : typeof colFilterMap[K] extends "number"
    ? NumberFilterType
    : typeof colFilterMap[K] extends "set"
    ? Set<Trade[K]>
    : never
}```